app.test.js

```diff
@@ -172,32 +172,62 @@ describe("GET /api/articles/:article_id/comments", () => {
   });
 });
 
-// describe.only("POST /api/articles/:article_id/comments", () => {
-//   test("Responds with posted comment", () => {
-//     newComment = { username: "rogersop", body: "nice article" };
-//     return request(app)
-//       .post("/api/articles/1/comments")
-//       .send(newComment)
-//       .expect(201)
-//       .then((res) => {
-//         expect(res.body.comment).toEqual({
-//           comment_id: 19,
-//           body: "nice article",
-//           article_id: 1,
-//           author: "rogersop",
-//           votes: expect.any(Number),
-//           created_at: expect.any(String),
-//         });
-//       });
-//   });
-  // test("Responds with posted comment", () => {
-  //   newComment = { username: "rogersop", body: "nice article" };
-  //   return request(app)
-  //     .post("/api/articles/0/comments")
-  //     .send(newComment)
-  //     .expect(404)
-  //     .then((res) => {
-  //       expect(res.body.msg).toBe("Article ID does not exist");
-  //     });
-  // });
-// })
+describe.only("POST /api/articles/:article_id/comments", () => {
+  test("Responds with posted comment", () => {
+    newComment = { username: "rogersop", body: "nice article" };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(201)
+      .then((res) => {
+        expect(res.body.comment).toEqual({
+          comment_id: 19,
+          body: "nice article",
+          article_id: 1,
+          author: "rogersop",
+          votes: expect.any(Number),
+          created_at: expect.any(String),
+        });
+      });
+  });
+  test("Returns 404 when posting to missing article ID number", () => {
+    newComment = { username: "rogersop", body: "nice article" };
+    return request(app)
+      .post("/api/articles/0/comments")
+      .send(newComment)
+      .expect(404)
+      .then((res) => {
+        expect(res.body.msg).toBe("Article ID not found");
+      });
+  });
+  test("Returns 400 if username property not present", () => {
+    newComment = { body: "nice article" };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(400)
+      .then((res) => {
+        expect(res.body.msg).toBe("Bad request");
+      });
+  });
+  test("Returns 400 if body property not present", () => {
+    newComment = { username: "rogersop" };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(400)
+      .then((res) => {
+        expect(res.body.msg).toBe("Bad request");
+      });
+  });
+  test("Returns 400 if wrong properties given", () => {
+    newComment = { username: "rogersop", wrongproperty: "badproperty" };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(400)
+      .then((res) => {
+        expect(res.body.msg).toBe("Bad request");
+      });
+  });
+})
```

Comment:
We still need to look at testing and handling for - 

- an invalid ID e.g. string of "not-an-id" instead of a number
- a username that does not exist